subprojects/dependency-management/src/main/java/org/gradle/internal/resolve/caching/ImplicitInputsCapturingInstantiator.java

```diff
@@ -42,6 +43,7 @@
  * method can still be called, in which case it creates a non capturing instance.
  *
  */
+@NonNullApi
```

Comment:
Why didn't you add `@NonNullApi` to the package info here?

Reply:
Because that would have resulted in a cascade of more nullability fixes. Gotta stop somewhere.